script/testing/junit/src/CreateIndexTest.java

```diff
@@ -0,0 +1,160 @@
+     * Create index tests
+     * ---------------------------------------------
+     */
+
+    /**
+     * Does a simple create index on a populated table
+     */
+    @Test
+    public void testSimpleCreate() throws SQLException {
+        String sql = "INSERT INTO tbl VALUES (1, 2, 100), (5, 6, 100), (3, 4, 100);";
+        Statement stmt = conn.createStatement();
+        stmt.execute(sql);
+        stmt.execute("CREATE INDEX tbl_ind ON tbl (c2)");
+        ResultSet rs = stmt.executeQuery("SELECT * FROM tbl WHERE c2 > 0 ORDER BY c2 ASC");
+        rs.next();
+        checkIntRow(rs, new String [] {"c1", "c2", "c3"}, new int [] {1, 2, 100});
+        rs.next();
+        checkIntRow(rs, new String [] {"c1", "c2", "c3"}, new int [] {3, 4, 100});
+        rs.next();
+        checkIntRow(rs, new String [] {"c1", "c2", "c3"}, new int [] {5, 6, 100});
+        assertNoMoreRows(rs);
+    }
+
+    /**
+     * Does a simple create index on a populated table
+     */
+    @Test
+    public void testWriteBlocking() throws SQLException, InterruptedException {
+        String sql = "INSERT INTO tbl VALUES (1, 2, 100), (5, 6, 100);";
+        Statement stmt = conn.createStatement();
+        int num_rows = 1500;
+        for (int i = 0; i < num_rows; i++) {
+            stmt.execute(sql);
+        }
+        Thread[] threads = new Thread[NUM_EXTRA_THREADS];
+        for (int i = 0; i < NUM_EXTRA_THREADS; i++) {
+            final Connection conn2 = thread_conn[i];
+            final int i2  = i;
+            thread_conn[i].setAutoCommit(i % 2 == 0);
+            threads[i] = new Thread(() -> {
+                try {
+                    Statement stmt2 = conn2.createStatement();
+                    for (int j = 0; j < num_rows; j++) {
+                        stmt2.execute("INSERT INTO tbl VALUES (3, 4, 200);");
+                    }
+                    if(i2 % 2 == 1) {
+                        conn2.commit();
+                    }
+                } catch(SQLException e) {
+                    DumpSQLException(e);
+                    Assert.fail();
+                }
+            });
+            threads[i].start();
+        }
+        Thread.sleep(100);
+        stmt.execute("CREATE INDEX tbl_ind ON tbl (c2)");
+        for (Thread t : threads) {
+            t.join();
+        }
+        ResultSet rs = stmt.executeQuery("SELECT * FROM tbl WHERE c2 > 0 ORDER BY c2 ASC");
+        for (int i = 0; i < num_rows; i++) {
+            rs.next();
+            checkIntRow(rs, new String [] {"c1", "c2", "c3"}, new int [] {1, 2, 100});
+        }
+        for (int i = 0; i < num_rows * NUM_EXTRA_THREADS; i++) {
+            rs.next();
+            checkIntRow(rs, new String [] {"c1", "c2", "c3"}, new int [] {3, 4, 200});
+        }
+        for (int i = 0; i < num_rows; i++) {
+            rs.next();
+            checkIntRow(rs, new String [] {"c1", "c2", "c3"}, new int [] {5, 6, 100});
+        }
+        assertNoMoreRows(rs);
+    }
+
+}
```

Comment:
Does this select statement go through the index?

Reply:
This has been verified manually - however without an EXPLAIN command implemented in the database we have unfortunately no way of programmatically verifying this